lib/interface.rb

```diff
@@ -0,0 +1,38 @@
+module Interface
+	def get_input
+	  puts "Please type a position to play your turn :"
+	  user_input = gets.chomp
+	  position = user_input.to_i
+  
+	  if user_input.to_i < 0 || user_input.to_i > 8
+			  puts "Thats not a valid position. Play again "
+			  position = get_input()
+	  end
+  
+	  if @board.taken?(user_input.to_i)
```

Comment:
You can simply use this: `if @board[value].is_a? Integer`

Thus, you don't need to declare that method anymore in the `Board` class